samples/boot/oauth2resourceserver-opaque/src/test/java/sample/OAuth2ResourceServerControllerTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package sample;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.springframework.security.test.oauth2.request.OAuth2MockMvcRequestPostProcessors.mockAccessToken;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.Collections;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ *
+ * @author Jérôme Wacongne &lt;ch4mp@c4-soft.com&gt;
+ * @since 5.2.0
+ *
+ */
+@RunWith(SpringRunner.class)
+@WebMvcTest(OAuth2ResourceServerController.class)
+public class OAuth2ResourceServerControllerTest {
```

Comment:
Could we update the `jwt` sample instead? I think it's just called `oauth2resourceserver`.

Reply:
This "opaque" sample is a good show-case that "jwt" PRs won't be enough for a serious OAuth2 test framework: testers must be empowered with the ability to configure `SecurityContext` with any kind of `Authentication` containing relevant types of `principal` (not `Principal`) so that `@AuthenticationPrincipal` retrieves an `Object` of the right type. This is the reason for me creating `access-token` and `oidcid` branches.

I updated the "jwt" sample once when introducing `jwt()` and again when introducing `@WithMockJwt`

Here it's "opaque" sample updated because of the new `accessToken()`. This opaque sample is further modified with `@WithMockAccessToken`

Reply:
as I remove access-token related devs from this PR, this test modification is gone also